opensrp-path/src/main/java/org/smartregister/path/service/intent/ValidateIntentService.java

```diff
@@ -0,0 +1,152 @@
+
+    private Context context;
+    private HTTPAgent httpAgent;
+    private static final int FETCH_LIMIT = 100;
+    private static final String VALIDATE_SYNC_PATH = "rest/validate/sync";
+
+
+    public ValidateIntentService() {
+        super("ValidateIntentService");
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        context = getBaseContext();
+        httpAgent = VaccinatorApplication.getInstance().context().getHttpAgent();
+        return super.onStartCommand(intent, flags, startId);
+
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+
+        try {
+
+             int fetchLimit = FETCH_LIMIT;
+            EventClientRepository db = VaccinatorApplication.getInstance().eventClientRepository();
+
+            List<String> clientIds = db.getUnValidatedClientBaseEntityIds(fetchLimit);
+            if (!clientIds.isEmpty()) {
+                fetchLimit -= clientIds.size();
+            }
+
+            List<String> eventIds = new ArrayList<>();
+            if (fetchLimit > 0) {
+                eventIds = db.getUnValidatedEventFormSubmissionIds(fetchLimit);
+            }
+
+            JSONObject request = request(clientIds, eventIds);
+            if (request == null) {
+                return;
+            }
+
+            String baseUrl = VaccinatorApplication.getInstance().context().configuration().dristhiBaseURL();
+            if (baseUrl.endsWith(context.getString(R.string.url_separator))) {
+                baseUrl = baseUrl.substring(0, baseUrl.lastIndexOf(context.getString(R.string.url_separator)));
+            }
+
+            String jsonPayload = request.toString();
+            Response<String> response = httpAgent.post(
+                    MessageFormat.format("{0}/{1}",
+                            baseUrl,
+                            VALIDATE_SYNC_PATH),
+                    jsonPayload);
+            if (response.isFailure()) {
+                Log.e(getClass().getName(), "Events sync failed.");
+                return;
+            }
+
+            JSONObject results = new JSONObject(response.payload());
+
+            if (results.has(getString(R.string.clients_key))) {
+                JSONArray inValidClients = results.getJSONArray(getString(R.string.clients_key));
+
+                for (int i = 0; i < inValidClients.length(); i++) {
+                    String inValidClientId = inValidClients.getString(i);
+                    clientIds.remove(inValidClientId);
+                    db.markEventValidationStatus(BaseRepository.TYPE_InValid, inValidClientId);
+                }
+
+                for (String clientId : clientIds) {
+                    db.markClientValidationStatus(BaseRepository.TYPE_Valid, clientId);
+                }
+            }
+
+            JSONArray inValidEvents = results.has("events") ? results.getJSONArray("events") : new JSONArray();
+            if (inValidEvents.length() > 0) {
```

Comment:
should be DRYd up line 100-110 duplicate lines 88-96, also don't need to check length

Reply:
So one loop is for clients and the other one for events